cli/azd/cmd/init.go

```diff
@@ -127,17 +124,12 @@ func newInitAction(
 }
 
 func (i *initAction) Run(ctx context.Context) (*actions.ActionResult, error) {
-	// In the case where `init` is run and a parent folder already has an `azure.yaml` file, the
-	// current ProjectDirectory will be set to that folder. That's not what we want here. We want
-	// to force using the current working directory as a project root (since we are initializing a
-	// new project).
 	wd, err := os.Getwd()
 	if err != nil {
 		return nil, fmt.Errorf("getting cwd: %w", err)
 	}
 
-	log.Printf("forcing project directory to %s", wd)
-	i.azdCtx.SetProjectDirectory(wd)
+	azdCtx := azdcontext.NewAzdContextWithDirectory(wd)
```

Comment:
Can you provide some more insight as to why we want to manually create the azd context here vs injection like in the other commands?

Reply:
I couldn't easily break off this change from #1296. This is a part of that. Once that lands, this diff will disappear.